app/matrix/matrix.component.ts

```diff
@@ -325,4 +359,132 @@ export class MatrixComponent implements OnInit, OnDestroy, AfterViewChecked {
   public parseUrl(url:string):any {
     return JSON.parse(`{"${url.replace(/&/g, '\",\"').replace(/=/g, '\":\"')}"}`);
   }
+
+  protected getCoords(querySelector:String, cb:any):any { // crossbrowser version
+    let box = this.element.querySelector(querySelector).getBoundingClientRect();
+
+    let body = document.body;
+    let docEl = document.documentElement;
+
+    let scrollTop = window.pageYOffset || docEl.scrollTop || body.scrollTop;
+    let scrollLeft = window.pageXOffset || docEl.scrollLeft || body.scrollLeft;
+
+    let clientTop = docEl.clientTop || body.clientTop || 0;
+    let clientLeft = docEl.clientLeft || body.clientLeft || 0;
+
+    let top = box.top;
+    let left = box.left + scrollLeft - clientLeft;
+
+    if (querySelector === '.images-container') {
+      top = box.top + scrollTop - clientTop;
+    }
+    cb({top: Math.round(top) + 40, left: Math.round(left) + 20});
+  }
+
+  protected startQuickTour():void {
+    this.closeOnboarding();
+    this.showOnboardingSwitcher = false;
+    this.numberOfStep = 1;
+    window.localStorage.setItem('onboarded', 'true');
+    setTimeout(() => {
+      this.getCoords('things-filter', (data:any) => {
+        this.baloonPosition = data;
+        this.switchOnQuickTour = true;
+        this.baloonName = this.baloonTips.thing.header;
+        this.baloonText = this.baloonTips.thing.description;
+      });
+    });
+  }
+
+  protected closeQuickTour():void {
+    this.showOnboardingSwitcher = true;
+    this.switchOnQuickTour = false;
+  }
+
+  protected step(step:boolean):void {
```

Comment:
Simplified by using on var instead of 6 vars.
Simplifying the function entirely seems to be useless because it makes to create one more var and removes only one string in every of 5 "if".